File: packages/storage/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/storage/classes/options/ListFilesOptions.java
package io.capawesome.capacitorjs.plugins.firebase.storage.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class ListFilesOptions {

    @NonNull
    private String path;

    private int maxResults;

    @Nullable
    private String pageToken;

    public ListFilesOptions(@NonNull String path, int maxResults, @Nullable String pageToken) {
        this.path = path;
        this.maxResults = maxResults;
        this.pageToken = pageToken;
    }

    @NonNull
    public String getPath() {
        return path;
    }

    public int getMaxResults() {
        return maxResults;
    }

    @Nullable
    public String getPageToken() {
        return pageToken;
    }
}

File: packages/storage/ios/Plugin/Classes/Options/ListFilesOptions.swift
import Foundation

@objc public class ListFilesOptions: NSObject {
    private var path: String
    private var maxResults: Int64
    private var pageToken: String?

    init(path: String, maxResults: Int, pageToken: String?) {
        self.path = path
        self.maxResults = Int64(maxResults)
        self.pageToken = pageToken
    }

    func getPath() -> String {
        return path
    }

    func getMaxResults() -> Int64 {
        return maxResults
    }

    func getPageToken() -> String? {
        return pageToken
    }
}
